lib/Test/Stream/Formatter/TAP.pm

```diff
@@ -99,6 +116,193 @@ sub _autoflush {
     select $old_fh;
 }
 
+sub event_tap {
+    my $self = shift;
+    my ($e, $num) = @_;
+
+    my $converter = $CONVERTERS{ref($e)};
+
+    $num = undef if $self->{+NO_NUMBERS};
+
+    # Legacy Support for $e->to_tap
+    unless ($converter) {
```

Comment:
This logic doesn't appear to expose what will happen if an event is seen without an associated converter.

It seems at present it handles such a case by pretending the event didn't exist, and I'm not sure that's the right choice.